tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/WafLibraryInvoker.cs

```diff
@@ -1,86 +1,50 @@
-// <copyright file="WafNative.cs" company="Datadog">
+// <copyright file="WafLibraryInvoker.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 using System;
-using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Datadog.Trace.AppSec.Waf.Initialization;
+using Datadog.Trace.AppSec.Waf.ReturnTypesManaged;
 using Datadog.Trace.Configuration;
 using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Serilog;
 
 namespace Datadog.Trace.AppSec.Waf.NativeBindings
 {
-    internal class WafNative
+    internal static class WafLibraryInvoker
```

Comment:
Be aware that by making this static, you can no longer unit test loading the library and handling different errors (I'm not sure if you were doing that already). Instead of making everything static, I think you could switch to the "singleton" approach we use elsewhere else, which would still allow you to test it while avoid issues of shared state between tests. Up to you, just pointing it out

Reply:
Yes I could do this too, true. Right now we can still unit tests it by calling `WafLibraryInvoker.Initialize` again and it overwrites the state of it..

Reply:
Be aware that different test classes run in _parallel_ in xunit, so you could get flakiness if multiple tests are calling `WafLibraryInvoker.Initialize` at the same time...

Reply:
yes they're all in the same xunit collection, but its true I will try and make it a singleton, it gives us more flexibility and it would be cleaner and will avoid headaches if we have parallel tests in the future, thanks!

Reply:
changed to singleton, well private field part of `Security` passed around , in 9daf1e931a478a25f20f4c693fc374289724d86d